### PR TITLE
Add css variable support and use MD icon

### PR DESCRIPTION
**Pull request by skjnldsv**
1. Migrate the icon to material design
2. Switch to css variable compatibility
3. Use `EmptyContent`
4. Use `getCapabilities`
5. use css to animate

Instant change on theming update:
Peek 31-03-2022 18-27

image

TODO in follow-up:
- [ ] Migrate the scss file to inner vue css file
- [ ] Cleanup old styling and vars usage

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -29,21 +42,23 @@
...
-						{{ t('notifications', 'No notifications') }}
-					</h2>
-				</div>
+
+				<!-- No notifications -->
+				<EmptyContent v-else>
+					{{ webNotificationsGranted === null
+						? t('notifications', 'Requesting browser permissions to show notifications')
+						: t('notifications', 'No notifications') }}
+					<template #icon><Bell /></template>
```
Comment (nickvergessen) [inline src/App.vue:60]:
```suggestion
					<template #icon><Bell title="" decorative /></template>
```

File: package.json [hunk 2]
```diff
@@ -21,11 +21,10 @@
 		"@nextcloud/capabilities": "^1.0.4",
 		"@nextcloud/dialogs": "^3.1.2",
 		"@nextcloud/event-bus": "^2.1.1",
-		"@nextcloud/initial-state": "^1.2.1",
```
Comment (nickvergessen) [inline package.json:24]: still needed in settings

**Comment by nickvergessen** [comment]
The icon of notifications are not correctly depending on the background anymore
And the empty content view looks a bit weird

Before | After
---|---
Bildschirmfoto von 2022-04-01 11-31-41 | Bildschirmfoto von 2022-04-01 11-30-34
Bildschirmfoto von 2022-04-01 11-31-58 | Bildschirmfoto von 2022-04-01 11-32-12

**Comment by skjnldsv** [comment]
> The icon of notifications are not correctly depending on the background anymore

How does that works?
How is the Notification API handling this?

**Comment by nickvergessen** [comment]
https://github.com/nextcloud/notifications/blob/master/css/styles.scss#L148-L150

I guess this does not match anymore

> How is the Notification API handling this?

The API can not handle it as the icon needs to be neutral for the mobile apps and desktop client so they can individually apply the dark mode behaviour.

**Comment by skjnldsv** [comment]
> [https://github.com/nextcloud/notifications/blob/master/css/styles.scss#L148-L150](https://github.com/nextcloud/notifications/blob/master/css/styles.scss?rgh-link-date=2022-04-04T20%3A25%3A42Z#L148-L150)
> 
> I guess this does not match anymore

Nice, I guess we can do this straight away in 25 once the scss cacher pr is merged

**Comment by skjnldsv** [comment]
Done @nickvergessen

**Review by szaimen — CHANGES_REQUESTED** [review]
Seems to work but didn't review the code.

Please adjust https://github.com/nextcloud/notifications/blob/6f9f9b50c30820f250d937a92a1128994b324e1d/css/styles.scss#L63 to `max-height: calc(100vh - 50px * 4 - 80px) !important;`

in order to fix 
image

**Comment by skjnldsv** [comment]
Fixed and fixed the animation too
Peek 22-04-2022 17-46

**Review by szaimen — APPROVED** [review]
Looks good from the screencast

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**